libraries/AP_Button/AP_Button.cpp

```diff
@@ -142,21 +142,14 @@ void AP_Button::timer_update(void)
  */
 void AP_Button::send_report(void)
 {
-    uint8_t chan_mask = GCS_MAVLINK::active_channel_mask();
-    uint32_t now = AP_HAL::millis();
-    for (uint8_t i=0; i<MAVLINK_COMM_NUM_BUFFERS; i++) {
-        if ((chan_mask & (1U<<i)) == 0) {
-            // not active
-            continue;
-        }
-        mavlink_channel_t chan = (mavlink_channel_t)i;
-        if (HAVE_PAYLOAD_SPACE(chan, BUTTON_CHANGE)) {
-            mavlink_msg_button_change_send(chan,
-                                           now,
-                                           (uint32_t)last_change_time_ms,
-                                           last_mask);
-        }
-    }
+    mavlink_message_t msg;
+    mavlink_msg_button_change_pack(mavlink_system.sysid,
```

Comment:
Sorry to crush your dreams, but this doesn't work. You are packing the message just for channel 0, which may have different proprieties to the channel the message actually ends up being sent to.

Reply:
Redone in terms of finalizing a `const char*` chunk of data on each channel.